### ISSUE TITLE
Can't download an OCI package twice if package contains symbolic links

### ISSUE THREAD

**Issue by flavioschuindt** [oras-project/oras#1593]
### What happened in your environment?

1) Created a package containing a symbolic link:

<img width="516" alt="image" src="https://github.com/user-attachments/assets/79b5c9b9-96bb-4f4a-9f42-f11a07ed4c59" />

2) Pushed it to my OCI complaint registry: ```oras push <MY_REPO>:<MY_TAG> ./src/:<MY_MEDIA_TYPE>``` where MY_REPO and MY_MEDIA_TYPE are redacted.

3) Pulled it one time and worked fine: ```oras pull <MY_REPO>:<MY_TAG>```

<img width="136" alt="image" src="https://github.com/user-attachments/assets/53f6bc74-9e40-4fb5-97b7-a2a491e47370" />

4) Without cleaning anything in the output folder, I tried the same command from (3) again, and then it failed:

```
Downloading 8b9900e74425 src
Error: failed to extract tar to <MY_OUTPUT_FOLDER>/src: symlink FILE1.md  <MY_OUTPUT_FOLDER>/src:FILE2.md: file exists
```

Correct me If I am wrong, but I believe what happens here is that internally oras is using tar to extract the compressed layers, but tar is refusing to overwrite a symbolic link.

### What did you expect to happen?

The second attempt should be successful and overwrites the files in the target output folder.

### How can we reproduce it?

Create an OCI package containing symbolic links and try the steps mentioned above.

### What is the version of your ORAS CLI?

Version:    1.1.0+Homebrew
Go version: go1.21.0

### What is your OS environment?

Mac OS Sonoma 14.4.1

### Are you willing to submit PRs to fix it?

- [ ] Yes, I am willing to fix it.

**Comment by ArghyaChakraborty** [comment]
I faced the exact same issue. Hopefully someone has some idea about how to avoid this ...

**Comment by alishah404** [comment]
+1
getting the same error, we have node modules in the OCI dir and node modules its failing with same error

```
Error: failed to extract tar to /tmp/oras-test/pkg: symlink ../acorn/bin/acorn /tmp/oras-test/pkg/node_modules/.bin/acorn: file exists
```

**Comment by Wwwsylvia** [comment]
Hi @flavioschuindt @ArghyaChakraborty @alishah404 , thank you for reporting this bug! We have opened an issue on the `oras-go` library side and will fix it there: https://github.com/oras-project/oras-go/issues/865

**Comment by flavioschuindt** [comment]
Thanks a lot, @Wwwsylvia. Help me understand one thing though: I see that in the ```oras-go``` the issue was labeled with the milestone v2.6.0. What is the usual release cadence, i.e., when should we expect v2.6.0 to be released? I am asking because we need to make an internal decision in our project, i.e., depending on the scheduled time we might need to implement an internal workaround, otherwise, we can wait and use the version with the fix which would be our preferred approach.

**Comment by Wwwsylvia** [comment]
@flavioschuindt Do you depend on `oras-go` or ORAS CLI? I suppose it's CLI?

For `oras-go`, we release new versions on-demand. Currently we don't have a clear target date for `v2.6.0` yet. But the hope is to get the items in the milestone done and release the new version ASAP.
For CLI, the next milestone is [v1.3.0-beta.2](https://github.com/oras-project/oras/milestone/21), which is supposed to be released by March. Once the bug is fixed on the oras-go side, CLI can uses the main branch without waiting for `oras-go` to release a tagged version.

**Comment by FeynmanZhou** [comment]
Maybe we could consider a patch release oras v2.5.1 to fix the symbolic link issue https://github.com/oras-project/oras-go/issues/865 since it impacts a lot of users. Let's discuss the oras-go milestone in the community meeting this Tuesday 4pm pacific time.

**Comment by FeynmanZhou** [comment]
@Wwwsylvia In addition to https://github.com/oras-project/oras-go/issues/865, is there any addition work in ORAS CLI to fix the symbolic link issue?

**Comment by Wwwsylvia** [comment]
@FeynmanZhou No, once the fix is merged into oras-go, CLI just needs to upgrade the go dependency.

**Comment by flavioschuindt** [comment]
> @flavioschuindt Do you depend on `oras-go` or ORAS CLI? I suppose it's CLI?
> 
> For `oras-go`, we release new versions on-demand. Currently we don't have a clear target date for `v2.6.0` yet. But the hope is to get the items in the milestone done and release the new version ASAP.
> For CLI, the next milestone is [v1.3.0-beta.2](https://github.com/oras-project/oras/milestone/21), which is supposed to be released by March. Once the bug is fixed on the oras-go side, CLI can uses the main branch without waiting for `oras-go` to release a tagged version. 
> 

My dependency is the oras CLI.

Thanks @FeynmanZhou for the patch suggestion. This is impacting not me, but other users as well, so implementing and releasing a patch as soon as possible seems to be a reasonable approach. How can I join the community meeting? The link you shared I only see agenda notes, is there a video call link or something that I can join?

**Comment by FeynmanZhou** [comment]
Hi @flavioschuindt , you can find the meeting schedule on https://hackmd.io/P-O6n222TcSMoJgHmTTduw?view#Zoom-Call-Info. We meet on Zoom. Feel free to join the meeting and connect with the community maintainers.
